packages/turbo/bin/turbo

```diff
@@ -1,10 +1,257 @@
+
+  if (otherInstalled.length > 0) {
+    console.error('Turborepo checked to see if binaries for another platform are installed.');
+    console.error('This typically indicates an error in sharing of pre-resolved node_modules across platforms.');
+    console.error('One common reason for this is copying files to Docker.');
+    console.error();
+    console.error(`We found these unnecessary binaries:`);
+    console.error(otherInstalled.join('\n'));
+  } else {
+    console.error(`We did not find any binaries on this system.`);
+    console.error(`This can happen if you run installation with the --no-optional flag.`);
+  }
+
+  // Check to see if we have partially-populated dependencies in the npm lockfile.
+  const MAX_LOOKUPS = 10;
+  const availablePackages = availablePlatforms.map(platform => availableArchs.map(arch => `turbo-${platform}-${arch}`)).flat();
+
+  try {
+    // Attempt to find project root.
+    const selfPath = require.resolve('turbo/package');
+
+    let previous = null;
+    let current = path.join(selfPath, '..', 'package-lock.json');
+
+    for (let i = 0; previous !== current && i < MAX_LOOKUPS; i++) {
+      try {
+        const lockfile = fs.readFileSync(current);
+        const parsedLockfile = JSON.parse(lockfile);
+
+        // If we don't show up in the lockfile it's the wrong lockfile.
+        if (parsedLockfile?.dependencies?.turbo) {
+          // Check to see if all of `turbo-<PLATFORM>-<ARCH>` is included.
+          const hasAllPackages = Object.keys(parsedLockfile?.dependencies ?? {}).filter(dependency => availablePackages.includes(dependency)).length === availablePackages.length;
+          if (!hasAllPackages) {
+            console.error();
+            console.error('***');
+            console.error();
+            console.error(`Turborepo detected that your lockfile (${current}) does not enumerate all available platforms.`);
+            console.error('This is likely a consequence of an npm issue: https://github.com/npm/cli/issues/4828.');
+
+            // Let's build their repair command:
+            let version = '';
+            let environment = '';
+            if (parsedLockfile?.packages[""]?.dependencies?.turbo) {
+              version = `@${parsedLockfile.packages[""].dependencies.turbo}`;
+              environment = ' --save-prod';
+            } else if (parsedLockfile?.packages[""]?.devDependencies?.turbo) {
+              version = `@${parsedLockfile.packages[""].devDependencies.turbo}`;
+              environment = ' --save-dev';
+            } else if (parsedLockfile?.packages[""]?.optionalDependencies?.turbo) {
+              version = `@${parsedLockfile.packages[""].optionalDependencies.turbo}`;
+              environment = ' --save-optional';
+            }
+
+            console.error();
+            console.error('To resolve this issue for your repository, run:');
+            console.error(`npm install turbo${version} --package-lock-only${environment} && npm install`);
```

Comment:
@chris-olszewski This is the magic command that fixes things without `rm -rf`.

Reply:
😮